app/views/user_sessions/_new.html.erb

```diff
@@ -38,4 +38,22 @@
     <p style="color: #888;"><%= raw t('user_sessions.new.reset_by_clicking_here', :url1 => "/reset/") %></p>
   </div>
   <% end %>
+  <% if !current_user %>
+      <script>
+        $(document).ready(function() {
+          $('a.requireLogin').click(function(e) {
+            e.preventDefault();
+            setMode("login");
```

Comment:
@JonathanXu1 Is this the line that is the fix? If so, could we maybe add it in this PR after #4264 is merged? Or should we could we just wait for this PR to be merged (including this)? Any thoughts @jywarren @SidharthBansal ?

This is looking awesome by the way!

Reply:
In #4264 you discussed that #4239 and #4228 needs a fix with your pr #4264. As at any stage we cannot merge any prs which will cause other prs to stop working. We should include your changes in this pr as they need some fixes suggested by @JonathanXu1. 
I hope it works well for both of you.
Thanks all.

Reply:
Yes I agree. We should wait for this to be merged before linking trying to link using the new class.

@JonathanXu1 Would you mind including my comments from #4264 in this section of the code for documentation?